extensions/amp-video/1.0/video-wrapper.js

```diff
@@ -0,0 +1,209 @@
+
+  const wrapperRef = useRef();
+  const playerRef = useRef();
+
+  const [userInteracted, setUserInteracted] = useState(false);
+  const readyDeferredRef = useRef(new Deferred());
+
+  const play = () => {
+    readyDeferredRef.current.promise.then(() => {
+      playerRef.current.play();
+    });
+  };
+
+  const pause = () => {
+    readyDeferredRef.current.promise.then(() => {
+      playerRef.current.pause();
+    });
+  };
+
+  useEffect(() => {
+    if (!mediasession || !metadata || !playing) {
+      return;
+    }
+    setMediaSession(window, metadata, play, pause);
+  }, [mediasession, playing, metadata]);
+
+  return (
+    <div
+      ref={wrapperRef}
+      style={{position: 'relative', width: '100%', height: '100%'}}
+    >
+      <Component
+        ref={playerRef}
+        // Ensure that props controlled here are excluded from passthroughProps,
+        // see destructuring statement above.
+        muted={muted}
+        controls={controls && (!autoplay || userInteracted)}
+        onLoad={() => readyDeferredRef.current.resolve()}
+        onLoadedMetadata={() => {
+          if (!mediasession) {
+            return;
+          }
+
+          const metadata = playerRef.current.getMetadata() || {
+            ...EMPTY_METADATA,
+            artwork: undefined,
+          };
+
+          metadata.title =
+            metadata.title ||
+            passthroughProps.title ||
+            passthroughProps['aria-label'] ||
+            document.title;
+
+          metadata.artwork =
+            metadata.artwork ||
+            parseSchemaImage(document) ||
+            parseOgImage(document) ||
+            parseFavicon(document);
+
+          setMetadata(metadata);
+        }}
+        onPlaying={() => setPlaying(true)}
+        onPause={() => setPlaying(false)}
+        {...passthroughProps}
+      >
+        {children}
+      </Component>
+      {autoplay && !userInteracted && (
+        <Autoplay
+          playing={playing}
+          displayIcon={!noaudio && muted}
+          wrapperRef={wrapperRef}
+          play={play}
+          pause={pause}
+          displayOverlay={controls}
+          onOverlayClick={() => {
+            setMuted(false);
+            setUserInteracted(false);
+          }}
+        />
+      )}
+    </div>
+  );
+}
+
+/**
+ * @param {!VideoAutoplayProps} props
+ * @return {PreactDef.Renderable}
+ */
+function Autoplay({
+  displayIcon,
+  playing,
+  displayOverlay,
+  onOverlayClick,
+  wrapperRef,
+  play,
+  pause,
+}) {
+  useEffect(() => {
+    let observer = new IntersectionObserver(
+      (entries) => {
+        if (entries[0].isIntersecting) {
+          play();
+        } else {
+          pause();
+        }
+      },
+      {threshold: MIN_VISIBILITY_RATIO_FOR_AUTOPLAY}
+    );
+
+    observer.observe(wrapperRef.current);
+
+    return () => {
+      observer.disconnect();
+      observer = null;
+    };
+  }, []);
+
+  return (
+    <>
+      {displayIcon && (
```

Comment:
Similar to current glyph implementation:

https://github.com/ampproject/amphtml/blob/c4a663d0ba13d0488c6fe73c55dc8c971ac6ec0d/src/service/video/autoplay.js#L47

Reply:
I'm realizing that we should flip this and get rid of `unpausable` instead. It's required only on older iOS versions.